README.md

```diff
@@ -33,6 +33,23 @@ Winit is designed to be a low-level brick in a hierarchy of libraries. Consequen
 show something on the window you need to use the platform-specific getters provided by winit, or
 another library.
 
+## Formatting Policy
+
+This repository strongly depends on the nightly version of `cargo-fmt`.
+A PR will not even get reviewed if nightly `cargo-fmt` check is not passed.
+
+Since this might be challenging to new rust users who don't know how to
+install and use nightly toolchain, here are the steps for reference:
+
+1. For installation, run `rustup toolchain install nightly`.
+2. For formatting, run `cargo +nightly fmt` at the repository root folder.
+
+Sidenotes:
+1. You may switch your default toolchain to nightly by running `rustup default nightly`.
+2. Formatting is the only expected scenario of using nightly toolchain in this repository.
```

Comment:
This is explained in rust book.

So should remove all the explanations here and just link to the relevant chapter in the book. I think it's this one https://rust-lang.github.io/rustup/concepts/toolchains.html

Reply:
I am afraid that this is still hostile to newcomers who are not intended to get involved into Rust too much.

Reply:
I'd just add a note like:

However, if you feel uncomfortable using `nightly` toolchain, ensure that CI passes and explicitly indicate that your code should be formatted before merge. Though, the CI should still pass, with the exception of formatting.

--

Though, we use nightly formatter for a very long time in alacritty, and I don't remember anyone struggling with it...